buildpack.md

```diff
@@ -719,26 +686,9 @@ Given the start command and execution strategy,
       2. Secondly, in alphabetically ascending order by layer directory name.
       3. Thirdly, in alphabetically ascending order by file name.
 
-1. If using an execution strategy involving a shell, the lifecycle MUST use a single shell process, with working directory `<app>`, to
-   1. source each file in each `<layers>/<layer>/profile.d` directory,
-      1. Firstly, in order of `/bin/build` execution used to construct the OCI image.
-      2. Secondly, in alphabetically ascending order by layer directory name.
-      3. Thirdly, in alphabetically ascending order by file name.
-   2. source each file in each `<layers>/<layer>/profile.d/<process>` directory,
-      1. Firstly, in order of `/bin/build` execution used to construct the OCI image.
-      2. Secondly, in alphabetically ascending order by layer directory name.
-      3. Thirdly, in alphabetically ascending order by file name.
-   3. source [†](README.md#linux-only)`<app>/.profile` or [‡](README.md#windows-only)`<app>/.profile.bat` if it is present.
-
-1. The lifecycle MUST set the working directory for the start command to `<working-dir>`, or to `<app>` if `<working-dir>` is not specified.
-
-1. The lifecycle MUST invoke the start command with the decided execution strategy.
-
-[†](README.md#linux-only)When executing a process using any execution strategy, the lifecycle SHOULD replace the lifecycle process in memory without forking it.
-
-[†](README.md#linux-only)When executing a process with Bash, the lifecycle SHOULD additionally replace the Bash process in memory without forking it.
+1. The lifecycle MUST use the `execve` syscall to invoke the command with its arguments, environment, and working directory following the process outlined in the [Platform Interface Specification](platform.md).
```

Comment:
The platform spec says:

> MUST execute the selected process as specified in the [Buildpack Interface Specfication](https://github.com/buildpacks/spec/blob/main/buildpack.md)


So I think something needs to change in the platform spec at least? But maybe we want a little more here? A potential division of details

Platform Spec describes:

* selection of user-defined process vs buildpack-defined process

* syntax for overriding args

Buildpack spec:

* Says that in the buildpack-defined process case: command+args will be executed but that args may be overridden with user provided args at the platform's discretion (basically what you have in the data format section below but moved up here so it's easier to find?)

Reply:
Would it be worth having a section for the process definition interface? Similar to what we have for [exec.d](https://github.com/buildpacks/spec/blob/main/buildpack.md#execd)?

Reply:
Good catch. tl;dr: I agree with the above, and I think we should:
- consolidate the execution details in platform spec, without pointer back to buildpack spec.
- add a table here with the command/args distinction.
- note here (maybe a footnote in the table) that it's at the platform's discretion.

Some more color:

I don't think the platform spec should point back to the buildpack spec about execution. The buildpack can express intention, but it is really up to the platform spec how to execute. Perhaps that is the main message that should go here...this is what these things mean, but it is up to the platform to execute.

For example, in implementing this RFC, when we change `command` from a string to a list of strings, platforms with API < 0.10 won't know that some args are override-able (`args`) and some aren't (`command[1:]`), as there is no way to express this distinction in the current metadata.toml format. (I walked through this in almost absurd detail in this miro board: https://miro.com/app/board/uXjVO8dPGUU=/?share_link_id=509704965141 )

Reply:
Took a shot at these items if you'd like to give it another pass.

Reply:
Love the table @mboldt 🎉  the absurd level of detail would be great for our docs! Maybe in the migration guide?